mapper/mapper.go

```diff
@@ -346,26 +348,52 @@ func createTableOfContents(documentSections []Section) coreModel.TableOfContents
 		},
 	}
 
-	sections := make(map[string]coreModel.ContentSection)
-	displayOrder := make([]string, 0)
-
-	for sectionIndex, section := range documentSections {
-		sectionId := fmt.Sprintf("section-%d", sectionIndex)
-		sections[sectionId] = coreModel.ContentSection{
-			Current: false,
-			Title: coreModel.Localisation{
-				Text: section.Title,
-			},
+	appendSections := func(list *[]Section, listType string, sections map[string]coreModel.ContentSection, displayOrder *[]string) {
+		for sectionIndex, section := range *list {
+			sectionId := fmt.Sprintf("%s-%d", listType, sectionIndex)
+			sections[sectionId] = coreModel.ContentSection{
+				Current: false,
+				Title: coreModel.Localisation{
+					Text: section.Title,
+				},
+			}
+			*displayOrder = append(*displayOrder, sectionId)
 		}
-		displayOrder = append(displayOrder, sectionId)
 	}
 
+	sections := make(map[string]coreModel.ContentSection)
+	displayOrder := make([]string, 0)
+	appendSections(&documentSections, "section", sections, &displayOrder)
+	appendSections(&documentAccordions, "accordion", sections, &displayOrder)
 	toc.Sections = sections
 	toc.DisplayOrder = displayOrder
-
 	return toc
 }
 
+type SectionReference struct {
+	Type  string
+	Index int
+}
+
+// Lookup section by the id used in the TableOfContents.DisplayOrder
+func (bulletin BulletinModel) FuncLookupSection(id string) (SectionReference, error) {
+	def := `(section|accordion)-(\d+)`
+	re := regexp.MustCompile(def)
+	matches := re.FindStringSubmatch(id)
+	if matches == nil {
+		err := fmt.Errorf(`unknown section type in id "%s"`, id)
+		return SectionReference{}, err
+	}
+	index, err := strconv.Atoi(matches[2])
```

Comment:
It would be better if you inform what `matches[2]` is so that it is easy to understand the data that it holds when reading the code.
You can do something like `???? := matches[2]` where ???? is whatever it is

Reply:
Were it not for the intervening string conversion, this line would be `index := matches[2]` which is exactly as you describe. You could go to extreme lengths to spell it out but it's redundant, because you know that `Atoi` is used to convert strings and you know from the capturing group in the regex that the match is a string, and that it's a string of numbers:

```
indexString := matches[2]
index, err := strconv.Atoi(indexString)
```

Reply:
Actually, you are completely right. In this case, it is very clear actually